Source/Blazorise/Components/Bar/Bar.razor.cs

```diff
@@ -115,14 +115,26 @@ public Task OnBreakpoint( bool broken )
         }
 
         /// <inheritdoc/>
-        protected override void Dispose( bool disposing )
+        protected override async ValueTask DisposeAsync( bool disposing )
         {
             if ( disposing )
             {
                 // make sure to unregister listener
                 if ( Rendered )
                 {
-                    _ = JSRunner.UnregisterBreakpointComponent( this );
+                    var task = JSRunner.UnregisterBreakpointComponent( this );
+
+                    try
+                    {
+                        await task;
+                    }
+                    catch
+                    {
+                        if ( !task.IsCanceled )
+                        {
+                            throw;
+                        }
+                    }
 
                     DisposeDotNetObjectRef( dotNetObjectRef );
```

Comment:
I think that we should call `DisposeDotNetObjectRef( dotNetObjectRef ); ` always regardless of exception or not.
When there is an exception, disposal of NetObjectRef is not called.
I'm not sure, if we need to catch task canceled exception there, it looks like that blazor does its own way.

maybee sth. like this?
```
try
{
    await task;
}
finally 
{
    DisposeDotNetObjectRef( dotNetObjectRef );
}
```

Reply:
Isn't the dispose of the NetObjectRef not required to unwire the javascript/c# bridge? So when the jsruntime times out isnt this all gone already?

Reply:
Do we want to just catch the exception and ignore it `try {} finally {}`, or do we still want to rethrow it `try {} catch { throw; } finally {}`?

If we ignore it then the app will hopefully go on and dispose of any other resources. In this case `DisposeDotNetObjectRef( dotNetObjectRef );`.